src/test/scala/wolfendale/scalacheck/regexp/GenParserSpec.scala

```diff
@@ -61,60 +58,32 @@ class GenParserSpec extends WordSpec with MustMatchers with PropertyChecks {
 
       forAll(gen) {
         a =>
-          GenParser.parse(s"[${a.mkString("")}]") mustEqual CharacterClass(
-            a.map(c => CharacterClass.Literal(c.toString)): _*
+          GenParser.parse(s"[${a.mkString("")}]") mustEqual CharacterClass.Group(
+            a.map(c => CharacterClass.Literal(c)): _*
           )
       }
     }
 
-    "parse a digit range" in {
-
-      val gen: Gen[(Int, Int)] =
-        for {
-          min <- Gen.chooseNum(0, 9)
-          max <- Gen.chooseNum(min, 9)
-        } yield (min, max)
-
-      forAll(gen) {
-        case (min, max) =>
-          GenParser.parse(s"[$min-$max]") mustEqual CharacterClass(CharacterClass.DigitRange(min, max))
-      }
-    }
-
-    "fail to parse an out of order digit range" ignore {
-
-      val gen: Gen[(Int, Int)] =
-        for {
-          min <- Gen.chooseNum(1, 9)
-          max <- Gen.chooseNum(0, min)
-        } yield (min, max)
-
-      forAll(gen) {
-        case (min, max) =>
-          an[Exception] mustBe thrownBy(GenParser.parse(s"[$min-$max]"))
-      }
-    }
-
-    "parse a lowercase char range" in {
+    "parse a char range" in {
 
       val gen: Gen[(Char, Char)] =
         for {
-          min <- Gen.alphaLowerChar
-          max <- Gen.choose(min, 'z')
+          min <- Gen.choose(Char.MinValue, Char.MaxValue)
+          max <- Gen.choose(min, Char.MaxValue)
         } yield (min, max)
 
       forAll(gen) {
         case (min, max) =>
-          GenParser.parse(s"[$min-$max]") mustEqual CharacterClass(CharacterClass.CharRange(min, max))
+          GenParser.parse(s"[$min-$max]") mustEqual CharacterClass.Group(CharacterClass.Range(min, max))
       }
     }
 
     "fail to parse an out of order lowercase char range" ignore {
```

Comment:
Currently test always passes. If you change:
```
an[Exception] mustBe thrownBy(GenParser.parse(s"[$min-$max"))
```
to
```
an[Exception] mustBe thrownBy(GenParser.parse(s"[$min-$max]"))
```
It will fail for the right reasons and you can remove `ignore`